launch/launch/descriptions/executable.py

```diff
@@ -0,0 +1,206 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# DISTRIBUTION A. Approved for public release; distribution unlimited.
+# OPSEC #4584.
+
+"""Module for a description of an Executable."""
+
+import os
+import re
+import shlex
+import threading
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from ..action import Action
+from ..launch_context import LaunchContext
+from ..some_substitutions_type import SomeSubstitutionsType
+from ..substitution import Substitution
+from ..substitutions import LaunchConfiguration
+from ..utilities import normalize_to_list_of_substitutions
+from ..utilities import perform_substitutions
+
+_executable_process_counter_lock = threading.Lock()
+_executable_process_counter = 0  # in Python3, this number is unbounded (no rollover)
+
+
+class Executable:
+    """Describes an executable (usually a single process) which may be run by the launch system."""
+
+    def __init__(
+        self, *,
+        cmd: Iterable[SomeSubstitutionsType],
+        prefix: Optional[SomeSubstitutionsType] = None,
+        name: Optional[SomeSubstitutionsType] = None,
+        cwd: Optional[SomeSubstitutionsType] = None,
+        env: Optional[Dict[SomeSubstitutionsType, SomeSubstitutionsType]] = None,
+        additional_env: Optional[Dict[SomeSubstitutionsType, SomeSubstitutionsType]] = None,
+        arguments: Optional[Iterable[SomeSubstitutionsType]] = None,
+    ) -> None:
+        """
+        Initialize an Executable description.
+
+        :param cmd: A list where the first item is the executable and the rest are
+            arguments to the executable, each item may be a string or a list of strings
+            and Substitutions to be resolved at runtime
+        :param prefix: a set of commands/arguments to preceed the cmd, used for
+            things like gdb/valgrind and defaults to the LaunchConfiguration
+            called 'launch-prefix'. Note that a non-default prefix provided in
+            a launch file will override the prefix provided via the `launch-prefix`
+            launch configuration regardless of whether the `launch-prefix-filter` launch
+            configuration is provided.
+        :param name: The label used to represent the process, as a string or a Substitution
+            to be resolved at runtime, defaults to the basename of the executable
+        :param cwd: The directory in which to run the executable
+        :param env: Dictionary of environment variables to be used, starting from a clean
+            environment. If None, the current environment is used.
+        :param additional_env: Dictionary of environment variables to be added. If env was
+            None, they are added to the current environment. If not, env is updated with
+            additional_env.
+        :param arguments: list of extra arguments for the executable
```

Comment:
I know this issue is quite old, but what's the purpose of this argument? Why could they not be part of the `cmd` argument?